### PR TITLE
gollama: init at 1.27.17

### LINKED ISSUE #355119: Package request: gollama
**Project description**

Gollama is a macOS / Linux tool for managing Ollama models.
image

**Metadata**

* homepage URL: https://github.com/sammcj/gollama?tab=readme-ov-file#installation
* source URL: https://github.com/sammcj/gollama?tab=readme-ov-file#installation
* license: MIT
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by genga898**
Closes #355119

Add [gollama](https://github.com/sammcj/gollama) to nixpkgs, a go tool for managing ollama models.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/gollama/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  buildGoModule,
+}:
+
+buildGoModule rec {
+  pname = "gollama";
+  version = "1.27.17";
+
+  src = fetchFromGitHub {
+    owner = "sammcj";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gollama/package.nix:13]:
```suggestion
    repo = "gollama";
```

File: pkgs/by-name/go/gollama/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-/KemOJwVHdb2BJnV53EVvbuE+0s3WOj4kzcox8hRZ6w=";
+  };
+
+  vendorHash = "sha256-MbadjPW9Oq3lRVa+Qcq4GXaZnBL0n6qLh5I2hJ0XhaY=";
+
+  doCheck = false;
+
+  ldFlags = [
+    "-s"
+    "-w"
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gollama/package.nix:24]: Do we have to override version? https://github.com/sammcj/gollama/blob/v1.27.17/main.go#L91

File: pkgs/by-name/go/gollama/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gollama/package.nix:26]: Would be better to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook

File: pkgs/by-name/go/gollama/package.nix (hunk 2 above)
Comment (genga898) [reply pkgs/by-name/go/gollama/package.nix:25]: Apologies, I don't fully understand the question.

Comment (genga898) [reply pkgs/by-name/go/gollama/package.nix:25]: It works

File: pkgs/by-name/go/gollama/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  versionCheckProgramArg = [ "-v" ];
+
+  meta = {
+    description = "Go manage your Ollama models";
+    homepage = "https://github.com/sammcj/gollama";
+    changelog = "https://github.com/sammcj/gollama/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ genga898 ];
+    mainProgram = "gollama";
```
Comment (adrlau) [inline pkgs/by-name/go/gollama/package.nix:40]: Shouldnt platforms be included in the meta 

```suggestion
    mainProgram = "gollama";
    platforms = platforms.linux ++ platforms.darwin;
```

Comment (genga898) [reply pkgs/by-name/go/gollama/package.nix:40]: It's not a must for them to be included

**Review by drupol — APPROVED** [review]

**Pull request merged**